hail/src/main/scala/is/hail/HailContext.scala

```diff
@@ -528,6 +613,50 @@ class HailContext private(
     }
   }
 
+  def readIndexedPartitions(
+    path: String,
+    rowsPath: String,
+    indexSpec: IndexSpec,
+    partFiles: Array[String],
+    intervalBounds: Option[Array[Interval]] = None
+  ): RDD[(InputStream, IndexReader, Option[Interval], InputMetrics)] = {
+    val idxPath = indexSpec.rel_path
+    val nPartitions = partFiles.length
+    val localHadoopConfBc = hadoopConfBc
+    val idxMetadata1 = IndexReader.readMetadata(
+      hadoopConf, s"$path/$idxPath/${ partFiles.head }.idx")
```

Comment:
this feels wrong. The index metadata should be in the `Spec`, I think.

Reply:
I agree. However, that's a future enhancement that probably requires redoing how we handle indices in a backwards incompatible way that would break BGEN indices.

Reply:
see my other comment -- I now think it should probably be in the RVDSpec.

Reply:
that seems like the right level for it